aptos-move/move-examples/veiled_coin/sources/veiled_coin.move

```diff
@@ -0,0 +1,669 @@
+        wrap_to<CoinType>(sender, signer::address_of(sender), amount)
+    }
+
+    /// Takes `amount` of `coin::Coin<CoinType>` coins from `sender`, wraps them inside a `VeiledCoin<CoinType>` and
+    /// sends to `recipient`. Note that the returned veiled coin will not actually be private yet, since the amount is public here.
+    public entry fun wrap_to<CoinType>(sender: &signer, recipient: address, amount: u64) acquires VeiledCoinMinter, VeiledCoinStore {
+        let c = coin::withdraw<CoinType>(sender, amount);
+
+        let vc = wrap_from_coin(c);
+
+        deposit<CoinType>(recipient, vc)
+    }
+
+    /// Takes `amount` of `VeiledCoin<CoinType>` coins from `sender`, unwraps them to a coin::Coin<CoinType>,
+    /// and sends them back to `sender`. Note that this function inherently leaks `amount`. Privacy of the remaining balance is maintained. Requires a range
+    /// proof on the new balance of the sender, to ensure the sender has enough money to send.
+    /// `amount` being the proper size is enforce by its being a u32.
+    public entry fun unwrap<CoinType>(
+	sender: &signer, 
+	amount: u32, 
+	range_proof_updated_balance: vector<u8>) acquires VeiledCoinStore, VeiledCoinMinter
+    {
+	unwrap_to<CoinType>(sender, signer::address_of(sender), amount, range_proof_updated_balance)
+    }
+
+    /// Takes `amount` of `VeiledCoin<CoinType>` coins from `sender`, unwraps them to a coin::Coin<CoinType>,
+    /// and sends them to `recipient`. It is not possible for this function to be made private. Requires a range
+    /// proof on the new balance of the sender, to ensure the sender has enough money to send.
+    /// `amount` being the proper size is enforce by its being a u32.
+    public entry fun unwrap_to<CoinType>(
+	sender: &signer, 
+	recipient: address, 
+	amount: u32, 
+	range_proof_updated_balance_bytes: vector<u8>) acquires VeiledCoinStore, VeiledCoinMinter
+    {
+	let range_proof_updated_balance = bulletproofs::range_proof_from_bytes(range_proof_updated_balance_bytes);
+	
+	let c = unwrap_to_coin<CoinType>(sender, amount, &range_proof_updated_balance);
+	coin::deposit<CoinType>(recipient, c);
+    }
+
+    /// Sends the specified private amount to `recipient` and updates the private balance of `sender`. Requires a range
+    /// proof on the new balance of the sender, to ensure the sender has enough money to send, in addition to a 
+    /// range proof on the transferred amount. Also requires a sigma protocol to prove that 
+    /// 'private_withdraw_amount' encrypts the same value using the same randomness as 'private_deposit_amount'. 
+    /// This value is the amount being transferred. These two ciphertexts are required as we need to update 
+    /// both the sender's and the recipient's balances, which use different public keys and so must be updated 
+    /// with ciphertexts encrypted with their respective public keys. 
+    public entry fun private_transfer_to_entry<CoinType>(
```

Comment:
`X` / `X_internal` feels weird to me because the internal functions are still going to be public and potentially used by other smart contracts, which makes them still external, just not entry functions. I'm not sure `_entry` is the right convention though